### ISSUE TITLE
The latest release 1.1.0 removed the latest training metrics from the result produced by Trainer.test method. Why?

### ISSUE THREAD

**Issue by pgagarinov** [PyTorchLightning/pytorch-lightning#5110]
For some reason Trainer.test methods doesn't return train metrics starting with v.1.1.0 PTL.
Was that done on purpose? If yes, then how can I get the latest values of training metrics?

v1.1.0:
--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{'test_acc': tensor(0.6441, device='cuda:0'),
 'test_acc_epoch': tensor(0.8204, device='cuda:0'),
 'test_f1': tensor(0.6441, device='cuda:0'),
 'test_f1_epoch': tensor(0.8204, device='cuda:0'),
 'test_loss': tensor(1.4021, device='cuda:0'),
 'test_loss_epoch': tensor(0.6918, device='cuda:0'),
 'test_prec': tensor(0.6441, device='cuda:0'),
 'test_prec_epoch': tensor(0.8204, device='cuda:0'),
 'test_rec': tensor(0.6441, device='cuda:0'),
 'test_rec_epoch': tensor(0.8204, device='cuda:0'),
 'val_acc': tensor(0.6441, device='cuda:0'),
 'val_acc_epoch': tensor(0.8204, device='cuda:0'),
 'val_f1': tensor(0.6441, device='cuda:0'),
 'val_f1_epoch': tensor(0.8204, device='cuda:0'),
 'val_loss': tensor(1.4021, device='cuda:0'),
 'val_loss_epoch': tensor(0.6918, device='cuda:0'),
 'val_prec': tensor(0.6441, device='cuda:0'),
 'val_prec_epoch': tensor(0.8204, device='cuda:0'),
 'val_rec': tensor(0.6441, device='cuda:0'),
 'val_rec_epoch': tensor(0.8204, device='cuda:0')

}

Before v1.1.0:
--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{'test_acc': tensor(0.6441, device='cuda:0'),
 'test_acc_epoch': tensor(0.8204, device='cuda:0'),
 'test_f1': tensor(0.6441, device='cuda:0'),
 'test_f1_epoch': tensor(0.8204, device='cuda:0'),
 'test_loss': tensor(1.4021, device='cuda:0'),
 'test_loss_epoch': tensor(0.6918, device='cuda:0'),
 'test_prec': tensor(0.6441, device='cuda:0'),
 'test_prec_epoch': tensor(0.8204, device='cuda:0'),
 'test_rec': tensor(0.6441, device='cuda:0'),
 'test_rec_epoch': tensor(0.8204, device='cuda:0'),
 'val_acc': tensor(0.6441, device='cuda:0'),
 'val_acc_epoch': tensor(0.8204, device='cuda:0'),
 'val_f1': tensor(0.6441, device='cuda:0'),
 'val_f1_epoch': tensor(0.8204, device='cuda:0'),
 'val_loss': tensor(1.4021, device='cuda:0'),
 'val_loss_epoch': tensor(0.6918, device='cuda:0'),
 'val_prec': tensor(0.6441, device='cuda:0'),
 'val_prec_epoch': tensor(0.8204, device='cuda:0'),
 'val_rec': tensor(0.6441, device='cuda:0'),
 'val_rec_epoch': tensor(0.8204, device='cuda:0'),
 'train_acc': 
 'train_acc_epoch': 
 'train_f1': 
 'train_f1_epoch': 
 'train_loss': 
 'train_loss_epoch': 
 'train_prec': 
 'train_prec_epoch': 
 'train_rec': 
 'train_rec_epoch': }

**Comment by tchaton** [comment]
Hey @pgagarinov,

Yes, when calling trainer.test(), only metrics computed around testing should be provided within the results objects.
To access all metrics, it can be done by using `trainer.callback_metrics`.

@edenlightning Do you second that ?

Best regards,
Thomas Chaton

**Comment by pgagarinov** [comment]
@tchaton Actually, it was very convenient to have all metrics (trainining, validation and test) in the same result to be able to compare them without having to call callback_metrics. Now it is less convenient((

**Comment by pgagarinov** [comment]
@tchaton Here is the strange thing -


 when I follow this line from the official documentation 
`
Metric objects can also be directly logged, in which case Lightning will log the metric based on on_step and on_epoch flags present in self.log(...). If on_epoch is True, the logger automatically logs the end of epoch metric value by calling .compute().
`
and  log metric objects (like Accuracy) directly trainer.callback_metrics returns the dictionary of tensors (I check from my custom callback)  until the training is finished. When the training is done callback_metrics returns a mix of tensors and objects like this:

Out[1]: 
{'train_acc_step': tensor(0.1094),
 'train_f1_step': tensor(0.0473),
 'train_prec_step': tensor(0.1103),
 'train_rec_step': tensor(0.1167),
 'train_loss_step': tensor(2.3015),
 'val_acc_epoch': tensor(0.0750),
 'val_f1_epoch': tensor(0.0219),
 'val_prec_epoch': tensor(0.0198),
 'val_rec_epoch': tensor(0.0937),
 'val_loss_epoch': tensor(2.3029),
 'val_acc': Accuracy(),
 'val_f1': F1(),
 'val_prec': Precision(),
 'val_rec': Recall(),
 'val_loss': tensor(2.3024),
 'train_acc_epoch': tensor(0.1094),
 'train_f1_epoch': tensor(0.0388),
 'train_prec_epoch': tensor(0.0755),
 'train_rec_epoch': tensor(0.1041),
 'train_loss_epoch': tensor(2.3010),
 'train_acc': Accuracy(),
 'train_f1': F1(),
 'train_prec': Precision(),
 'train_rec': Recall(),
 'train_loss': tensor(2.3015)}

Is this an expected behavior?
